app/Http/Controllers/Core/V1/Organisation/ImportController.php

```diff
@@ -0,0 +1,213 @@
+        }
+        if (!$file_blob = base64_decode(trim($matches[2]), true)) {
+            throw new ValidationException('Invalid Base64 Excel data');
+        }
+        return $this->storeBinaryUpload($file_blob, $path, $matches[1]);
+    }
+
+    /**
+     * Get the mime type of a binary file string
+     *
+     * @var String $file_str
+     * @return String mime type
+     **/
+    public function getFileStringMimeType(string $file_str)
+    {
+        $finfo = finfo_open(FILEINFO_MIME_TYPE);
+        $mime_type = finfo_buffer($finfo, $file_str);
+        finfo_close($finfo);
+        return $mime_type;
+    }
+
+    /**
+     * Guess the extension for a file from it's mime-type
+     *
+     * @param String $mime_type
+     * @return String
+     **/
+    public function guessFileExtension(string $mime_type)
+    {
+        return (new MimeTypes)->getExtensions($mime_type)[0] ?? null;
+    }
+
+    /**
+     * Validate the spreadsheet rows
+     *
+     * @param String $filePath
+     * @return Array
+     **/
+    public function validateSpreadsheet(String $filePath)
+    {
+        $spreadsheetHandler = new SpreadsheetHandler();
+
+        $spreadsheetHandler->import(Storage::disk('local')->path($filePath));
+
+        $spreadsheetHandler->readHeaders();
+
+        $rejectedRows = [];
+
+        foreach ($spreadsheetHandler->readRows() as $i => $row) {
+            $validator = Validator::make($row, [
+                'name' => ['required', 'string', 'min:1', 'max:255'],
+                'description' => ['required', 'string', 'min:1', 'max:10000'],
+                'url' => ['present', 'url', 'max:255'],
+                'email' => ['present', 'nullable', 'required_without:phone', 'email', 'max:255'],
+                'phone' => [
+                    'present',
+                    'nullable',
+                    'required_without:email',
+                    'string',
+                    'min:1',
+                    'max:255',
+                ],
+            ]);
+
+            if ($validator->fails()) {
+                $row['index'] = $i;
+                $rejectedRows[] = ['row' => $row, 'errors' => $validator->errors()];
+            }
+        }
+
+        return $rejectedRows;
+    }
+
+    /**
+     * Import the uploaded file contents
+     *
+     * @param String $filePath
+     * @return null
+     **/
+    public function importSpreadsheet(String $filePath)
+    {
+        $spreadsheetHandler = new SpreadsheetHandler();
+
+        $spreadsheetHandler->import(Storage::disk('local')->path($filePath));
+
+        $spreadsheetHandler->readHeaders();
+
+        $importedRows = 0;
+
+        DB::transaction(function () use ($spreadsheetHandler, &$importedRows) {
+            $rowBatch = [];
+            foreach ($spreadsheetHandler->readRows() as $row) {
+                $row['id'] = (string) Str::uuid();
+                $row['slug'] = Str::slug($row['name'] . ' ' . uniqid(), '-');
+                $rowBatch[] = $row;
```

Comment:
Can we add the `created_at` and `updated_at` timestamps?

```suggestion
                $row['id'] = (string) Str::uuid();
                $row['slug'] = Str::slug($row['name'] . ' ' . uniqid(), '-');
                $row['created_at'] = Date::now();
                $row['updated_at'] = Date::now();
                $rowBatch[] = $row;
```

Reply:
Done.